File: coko/com.elster.jupiter.issue.extjs/src/main/web/js/issue/src/model/Action.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Isu.model.Action', {
    extend: 'Ext.data.Model',
    requires: [
        'Uni.property.model.Property'
    ],
    fields: [
        {
            name: 'id',
            type: 'int'
        },
        {
            name: 'name',
            type: 'text'
        },
        {
            name: 'issueType',
            type: 'auto'
        }
    ],
    associations: [
        {
            name: 'properties',
            type: 'hasMany',
            model: 'Uni.property.model.Property',
            associationKey: 'properties',
            foreignKey: 'properties'
        }
    ],
    proxy: {
        type: 'rest',
        url: '/api/isu/actions',
        reader: {
            type: 'json'
        }
    }
});

File: coin/com.elster.insight.usagepoint.data/src/main/java/com/elster/jupiter/mdm/usagepoint/data/Action.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.mdm.usagepoint.data;

public enum Action {
    ADD_CALENDAR("addCalendar");

    private final String matcherString;

    Action(String matcherString) {
        this.matcherString = matcherString;
    }

    public boolean matches(String action) {
        return matcherString.equalsIgnoreCase(action);
    }
}
